todo_project_name/rsa.py

```diff
@@ -6,21 +6,46 @@
 from dataclasses import dataclass
 from .md4 import MD4
 from .md5 import MD5
+from abc import ABC
+
+
+class RSAKey(ABC):
+    def __init__(self, key: int, modulus: int, id: Optional[str] = None) -> None:
+        """Create a new instance.
+
+        Parameters
+        ==========
+        id:
+            cannot have white space at its ends or be an empty string.
+        """
+        if id is not None:
+            if id == "":
+                raise ValueError("The string cannot be empty.")
+            if id != id.strip():
+                # TODO: Consider lifting this constraint with a more flexible serialization.
+                # Is it needed?
+                raise ValueError("`id` cannot have white space at its ends.")
+
+        self.key = key
+        self.modulus = modulus
+        self.id = id
+
+    def __repr__(self) -> str:
+        name = self.__class__.__name__
+        return f"{name}(key={self.key!r}, modulus={self.modulus!r}, id={self.id!r})"
+
+    def __eq__(self, other) -> bool:
```

Comment:
Id is irrelevant?

Reply:
To the comparison? I believe, yes. They are not the same object, obviously, but they allow for the same things in application. No matter which equal object you choose, you can encrypt or decrypt the same data.